stdlib/src/collections/list.mojo

```diff
@@ -568,8 +568,6 @@ struct List[T: CollectionElement](CollectionElement, Sized, Boolable):
             self.resize(new_size)
         else:
             self.reserve(new_size)
-            for i in range(new_size, self.size):
```

Comment:
Your change is correct, but it was not a bug. Those two lines were actually useless and no-op. If `new_size > self.size` then the range is null. Thus the `destroy_pointee()` can never be executed.
You can try this in a standalone script:
```mojo
def main():
    for i in range(10, 5):
        print(i)  # <- never executed
    print("done")
```

Reply:
Agreed. Still great to remove these useless lines those, thanks @rd4com!